client/apps/edit2/components/content/section_controls/layout.jsx

```diff
@@ -0,0 +1,123 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { IconImageFullscreen } from '@artsy/reaction-force/dist/Components/Publishing'
+import { onChangeSection } from 'client/actions/editActions'
+
+export class LayoutControls extends Component {
+  static propTypes = {
+    article: PropTypes.object,
+    channel: PropTypes.object,
+    disabledAlert: PropTypes.func,
+    section: PropTypes.object,
+    onChange: PropTypes.func
+  }
+
+  changeLayout = (layout) => {
+    const { section, disabledAlert, onChange } = this.props
+    const { images, type } = section
+
+    const isFillwidth = layout === 'fillwidth'
+    const isImage = this.sectionIsImage()
+
+    if (isFillwidth && isImage && images.length > 1) {
+      return disabledAlert()
+    }
+    if (type === 'image_set') {
+      onChange('type', 'image_collection')
+    }
+    onChange('layout', layout)
+  }
+
+  toggleImageSet = () => {
+    const { section, onChange } = this.props
+
+    if (section.type === 'image_collection') {
+      onChange('type', 'image_set')
+      onChange('layout', 'mini')
```

Comment:
Is it necessary to fire two back-to-back actions (leading to two render cycles), or could this be batched like 

```javascript
onChange({
  type: 'image_set',
  layout: 'mini'
})
```

Reply:
This is on my list of TODOs, but was not planning on changing in this PR.